### PR TITLE
fix(vue): decorator event props not work in vue2

### LINKED ISSUE #3883: [Bug Report] @formily/vue x-decorator-props @click event not work in vue2
- [x] I have searched the [issues](https://github.com/alibaba/formily/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/formily-vue-decoratorprops-event-not-work-in-vue2-wndtty?file=/src/App.vue)

### Steps to reproduce
Use @formily/element or other Vue UI library with **"x-decorator-props": { onClick: ..., @update: ... }**

use vue2 environment please.

### What is expected?
event in decorator props will work.

### What is actually happening?
In Vue3, events in decorator props work normally

See: https://codesandbox.io/s/formily-vue-decoratorprops-event-work-in-vue3-w42hm5?file=/src/App.vue:0-1914

But in Vue2, events in decorator props not work as expected

See: https://codesandbox.io/s/formily-vue-decoratorprops-event-not-work-in-vue2-wndtty?file=/src/App.vue

### Package
@formily/vue@2.2.26

---

Bug Reason:

In code: https://github.com/alibaba/formily/blob/f51283430e7723510dc82339c43297b83ed59c1f/packages/vue/src/components/ReactiveField.ts#L159-L173

decoratorProps will become { attrs: ... }, it will be infer to props/attrs/events in vue3,  but in vue2 attrs cannot become events automatically.

It should deal with like this snippet code https://github.com/alibaba/formily/blob/f51283430e7723510dc82339c43297b83ed59c1f/packages/vue/src/components/ReactiveField.ts#L183-L202

### PR DESCRIPTION
**Pull request by MeetzhDing**
_Before_ submitting a pull request, please make sure the following is done...

- [x] Ensure the pull request title and commit message follow the [Commit Specific](https://formilyjs.org/guide/contribution#pr-specification) in **English**.
- [x] Fork the repo and create your branch from `master` or `formily_next`.
- [x] If you've added code that should be tested, add tests!
- [x] If you've changed APIs, update the documentation.
- [x] Ensure the test suite passes (`npm test`).
- [x] Make sure your code lints (`npm run lint`) - we've done our best to make sure these rules match our internal linting guidelines.

**Please do not delete the above content**

---

## What have you changed?
fix https://github.com/alibaba/formily/issues/3883

By the way, all build/test work failed in my computer,  it seems was a monorepo problem, but i don't know how to slove it.

### CODE REVIEW

**Comment by janryWang** [comment]
你需要先全量构建才能跑起来，还有node环境，最好用12

**Comment by MeetzhDing** [comment]
@janryWang please check this pr again. all test (include new one) pass .

Can I get a milestone about this fix?

File: packages/vue/src/components/ReactiveField.ts [hunk 1]
```diff
@@ -160,10 +160,26 @@ export default observer({
           FormPath.getIn(options?.components, field.decoratorType as string) ??
           field.decoratorType
         const componentAttrs = toJS(field.decorator[1]) || {}
+
+        const events: Record<string, any> = {}
+        Object.keys(componentAttrs)
```
Comment (janryWang) [inline packages/vue/src/components/ReactiveField.ts]: 优化一下性能吧，可以用 shared 里的 each，一次遍历搞定，这个遍历次数太多了

Comment (MeetzhDing) [reply packages/vue/src/components/ReactiveField.ts]: all done. @janryWang

**Pull request merged**